### PR TITLE
Tools: improve manifest generation

### PR DESCRIPTION
**Pull request by tridge**
This improves the manifest generation by listing all possible USB VID/PID pairs each fw can have, plus the board_id and bootloader strings.
This will make it possible for a GCS to offer a list of matching firmwares. This is important for boards such as fmuv5 where one bootloader is used for multiple distinct boards

This is what we end up with:
http://uav.tridgell.net/Manifest/

### CODE REVIEW

File: Tools/scripts/build_binaries.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 import time
 import subprocess
 import sys
-import zlib
+import zlib, gzip
```
Comment (peterbarker) [inline Tools/scripts/build_binaries.py]: flake8 will bitch about this.

`build_binaries.py` is currently flake8-clean.  Near-enough:


```
pbarker@bluebottle:~/rc/ardupilot(rally-protocol)$ flake8 Tools/scripts/build_binaries.py 
Tools/scripts/build_binaries.py:161:44: E231 missing whitespace after ','
Tools/scripts/build_binaries.py:162:44: E231 missing whitespace after ','
Tools/scripts/build_binaries.py:164:28: E713 test for membership should be 'not in'
Tools/scripts/build_binaries.py:361:37: E703 statement ends with a semicolon
Tools/scripts/build_binaries.py:415:12: E713 test for membership should be 'not in'
Tools/scripts/build_binaries.py:667:29: E703 statement ends with a semicolon
pbarker@bluebottle:~/rc/ardupilot(rally-protocol)$ 
```

File: Tools/scripts/generate_manifest.py [hunk 2]
```diff
@@ -257,6 +370,7 @@ def walk_directory(self, basedir):
                         if fulldate in ["files.html", ".makehtml"]:
                             # generated file which should be ignored
                             continue
+                        print(fulldate, file=sys.stderr)
```
Comment (peterbarker) [inline Tools/scripts/generate_manifest.py]: stray debug?

File: Tools/scripts/generate_manifest.py [hunk 3]
```diff
@@ -214,11 +324,12 @@ def xfirmwares_to_firmwares(self, xfirmwares):
         else:
             return [xfirmwares]
 
-    known_release_types = {
-        "beta": 1,
-        "latest": 1,
-        "stable": 1
-    }
+    def check_release_type(self, tag):
```
Comment (peterbarker) [inline Tools/scripts/generate_manifest.py]: How about `known_release_type` or `valid_release_type`?

File: Tools/scripts/build_binaries.py (hunk 1 above)
Comment (tridge) [reply Tools/scripts/build_binaries.py]: fixed

File: Tools/scripts/generate_manifest.py (hunk 2 above)
Comment (tridge) [reply Tools/scripts/generate_manifest.py]: fixed

File: Tools/scripts/generate_manifest.py (hunk 3 above)
Comment (tridge) [reply Tools/scripts/generate_manifest.py]: fixed

**Pull request merged**